pyphare/pyphare/pharesee/hierarchy/tensorfield.py

```
@@ -0,0 +1,193 @@
+
+    @staticmethod
+    def FROM(cls, hier):
+        return cls(
+            hier.patch_levels,
+            hier.domain_box,
+            hier.refinement_ratio,
+            hier.times(),
+            hier.data_files,
+        )
+
+    def __getitem__(self, input):
```

Ruff (0.15.10)

[error] Function argument `input` is shadowing a Python builtin

(A002)